from PyQt5 import QtCore, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(500, 600)
        MainWindow.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.verticalLayoutWidget_2 = QtWidgets.QWidget(self.centralwidget)
        self.verticalLayoutWidget_2.setGeometry(QtCore.QRect(10, 10, 750, 500))
        self.verticalLayoutWidget_2.setObjectName("verticalLayoutWidget_2")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.verticalLayoutWidget_2)
        self.verticalLayout_2.setSizeConstraint(QtWidgets.QLayout.SetDefaultConstraint)
        self.verticalLayout_2.setContentsMargins(0, 0, 0, 0)
        self.verticalLayout_2.setSpacing(6)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.horizontalLayout = QtWidgets.QHBoxLayout()
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.label = QtWidgets.QLabel(self.verticalLayoutWidget_2)
        self.label.setObjectName("label")
        self.horizontalLayout.addWidget(self.label, 0, QtCore.Qt.AlignLeft)
        self.lcdNumber = QtWidgets.QLCDNumber(self.verticalLayoutWidget_2)
        self.lcdNumber.setObjectName("lcdNumber")
        self.lcdNumber.setFixedSize(50, 30)
        self.horizontalLayout.addWidget(self.lcdNumber, 0, QtCore.Qt.AlignLeft)
        spacerItem = QtWidgets.QSpacerItem(40, 20, QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        self.horizontalLayout.addItem(spacerItem)
        self.verticalLayout_2.addLayout(self.horizontalLayout)
        self.gameFieldTableView = QtWidgets.QTableView(self.verticalLayoutWidget_2)
        self.gameFieldTableView.setStyleSheet("QTableView { gridline-color: green; }")
        self.gameFieldTableView.setFixedSize(422, 362)
        self.gameFieldTableView.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.gameFieldTableView.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
        self.gameFieldTableView.setSelectionMode(QtWidgets.QAbstractItemView.NoSelection)
        self.gameFieldTableView.setShowGrid(False)
        self.gameFieldTableView.setObjectName("gameFieldTableView")
        self.gameFieldTableView.horizontalHeader().setVisible(False)
        self.gameFieldTableView.horizontalHeader().setDefaultSectionSize(30)
        self.gameFieldTableView.verticalHeader().setDefaultSectionSize(30)
        self.gameFieldTableView.verticalHeader().setVisible(False)
        self.verticalLayout_2.addWidget(self.gameFieldTableView)
        self.colorButtonsTableView = QtWidgets.QTableView(self.verticalLayoutWidget_2)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.colorButtonsTableView.sizePolicy().hasHeightForWidth())
        self.colorButtonsTableView.setSizePolicy(sizePolicy)
        self.colorButtonsTableView.setFixedSize(302, 50)
        self.colorButtonsTableView.setSelectionMode(QtWidgets.QAbstractItemView.NoSelection)
        self.colorButtonsTableView.setObjectName("tableView")
        self.colorButtonsTableView.horizontalHeader().setVisible(False)
        self.colorButtonsTableView.verticalHeader().setVisible(False)
        self.colorButtonsTableView.setStyleSheet("QTableView { gridline-color: green; }")
        self.colorButtonsTableView.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.colorButtonsTableView.setShowGrid(True)
        self.colorButtonsTableView.horizontalHeader().setDefaultSectionSize(50)
        self.colorButtonsTableView.verticalHeader().setDefaultSectionSize(50)
        self.verticalLayout_2.addWidget(self.colorButtonsTableView)
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 691, 21))
        self.menubar.setObjectName("menubar")
        self.menuGame = QtWidgets.QMenu(self.menubar)
        self.menuGame.setObjectName("menuGame")
        self.menuAbout = QtWidgets.QMenu(self.menubar)
        self.menuAbout.setObjectName("menuAbout")
        MainWindow.setMenuBar(self.menubar)
        self.actionNewGame = QtWidgets.QAction(MainWindow)
        self.actionNewGame.setObjectName("actionNewGame")
        self.actionSettings = QtWidgets.QAction(MainWindow)
        self.actionSettings.setObjectName("actionSettings")
        self.separator = QtWidgets.QAction(MainWindow)
        self.separator.setObjectName("separator")
        self.actionExit = QtWidgets.QAction(MainWindow)
        self.actionExit.setObjectName("actionExit")
        self.actionAboutGame = QtWidgets.QAction(MainWindow)
        self.actionAboutGame.setObjectName("action_9")
        self.actionAboutAuthor = QtWidgets.QAction(MainWindow)
        self.actionAboutAuthor.setObjectName("action_10")
        self.menuGame.addAction(self.actionNewGame)
        self.menuGame.addAction(self.actionSettings)
        self.menuGame.addSeparator()
        self.menuGame.addAction(self.actionExit)
        self.menuAbout.addAction(self.actionAboutGame)
        self.menuAbout.addAction(self.actionAboutAuthor)
        self.menubar.addAction(self.menuGame.menuAction())
        self.menubar.addAction(self.menuAbout.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "Перекраска"))
        self.label.setText(_translate("MainWindow", "Количество ходов:"))
        self.menuGame.setTitle(_translate("MainWindow", "Игра"))
        self.menuAbout.setTitle(_translate("MainWindow", "Справка"))
        self.actionNewGame.setText(_translate("MainWindow", "&Новая"))
        self.actionSettings.setText(_translate("MainWindow", "&Параметры"))
        self.separator.setText(_translate("MainWindow", "-"))
        self.actionExit.setText(_translate("MainWindow", "&Выход"))
        self.actionAboutGame.setText(_translate("MainWindow", "Об игре"))
        self.actionAboutAuthor.setText(_translate("MainWindow", "Об авторе"))
